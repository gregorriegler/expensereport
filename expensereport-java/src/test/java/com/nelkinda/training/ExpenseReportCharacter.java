package com.nelkinda.training;

import org.approvaltests.ApprovalUtilities;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.util.Date;

@UseReporter(IdeaReporter.class)
class ExpenseReportCharacter {
    
    @Test
    void noExpenses() {
        ByteArrayOutputStream output = printExpenses(new Expenses());
        Approvals.verify(output);
    }

    @Test
    public void manyExpenses() {
        ByteArrayOutputStream output = printExpenses(new Expenses(
            Expense.createCarRentalExpense(7),
            Expense.createBreakfastExpense(0),
            Expense.createBreakfastExpense(1),
            Expense.createBreakfastExpense(1000),
            Expense.createBreakfastExpense(1001),
            Expense.createDinnerExpense(5000),
            Expense.createDinnerExpense(5001),
            Expense.createLunchExpense(2000),
            Expense.createLunchExpense(2001)
        ));
        Approvals.verify(output);
    }

    private ByteArrayOutputStream printExpenses(Expenses expenses) {
        ByteArrayOutputStream output = new ApprovalUtilities().writeSystemOutToStringBuffer();
        ExpenseReport report = new ExpenseReport() {
            @Override
            protected Date now() {
                return new Date(0);
            }
        };
        report.printReport(expenses);
        return output;
    }
}