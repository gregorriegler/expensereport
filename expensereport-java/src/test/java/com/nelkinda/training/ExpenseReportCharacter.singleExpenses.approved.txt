[DINNER, -1] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	-1	 
Meal expenses: -1
Total expenses: -1
 
[DINNER, 0] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	0	 
Meal expenses: 0
Total expenses: 0
 
[DINNER, 500] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	500	 
Meal expenses: 500
Total expenses: 500
 
[DINNER, 999] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	999	 
Meal expenses: 999
Total expenses: 999
 
[DINNER, 1000] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	1000	 
Meal expenses: 1000
Total expenses: 1000
 
[DINNER, 1001] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	1001	 
Meal expenses: 1001
Total expenses: 1001
 
[DINNER, 4999] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	4999	 
Meal expenses: 4999
Total expenses: 4999
 
[DINNER, 5000] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	5000	 
Meal expenses: 5000
Total expenses: 5000
 
[DINNER, 5001] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	5001	X
Meal expenses: 5001
Total expenses: 5001
 
[DINNER, 10000] => Expenses Thu Jan 01 01:00:00 CET 1970
Dinner	10000	X
Meal expenses: 10000
Total expenses: 10000
 
[BREAKFAST, -1] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	-1	 
Meal expenses: -1
Total expenses: -1
 
[BREAKFAST, 0] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	0	 
Meal expenses: 0
Total expenses: 0
 
[BREAKFAST, 500] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	500	 
Meal expenses: 500
Total expenses: 500
 
[BREAKFAST, 999] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	999	 
Meal expenses: 999
Total expenses: 999
 
[BREAKFAST, 1000] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	1000	 
Meal expenses: 1000
Total expenses: 1000
 
[BREAKFAST, 1001] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	1001	X
Meal expenses: 1001
Total expenses: 1001
 
[BREAKFAST, 4999] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	4999	X
Meal expenses: 4999
Total expenses: 4999
 
[BREAKFAST, 5000] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	5000	X
Meal expenses: 5000
Total expenses: 5000
 
[BREAKFAST, 5001] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	5001	X
Meal expenses: 5001
Total expenses: 5001
 
[BREAKFAST, 10000] => Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	10000	X
Meal expenses: 10000
Total expenses: 10000
 
[CAR_RENTAL, -1] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	-1	 
Meal expenses: 0
Total expenses: -1
 
[CAR_RENTAL, 0] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	0	 
Meal expenses: 0
Total expenses: 0
 
[CAR_RENTAL, 500] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	500	 
Meal expenses: 0
Total expenses: 500
 
[CAR_RENTAL, 999] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	999	 
Meal expenses: 0
Total expenses: 999
 
[CAR_RENTAL, 1000] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	1000	 
Meal expenses: 0
Total expenses: 1000
 
[CAR_RENTAL, 1001] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	1001	 
Meal expenses: 0
Total expenses: 1001
 
[CAR_RENTAL, 4999] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	4999	 
Meal expenses: 0
Total expenses: 4999
 
[CAR_RENTAL, 5000] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	5000	 
Meal expenses: 0
Total expenses: 5000
 
[CAR_RENTAL, 5001] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	5001	 
Meal expenses: 0
Total expenses: 5001
 
[CAR_RENTAL, 10000] => Expenses Thu Jan 01 01:00:00 CET 1970
Car Rental	10000	 
Meal expenses: 0
Total expenses: 10000
 
